src/ape/contracts/base.py

```diff
@@ -48,6 +48,17 @@ def __init__(
     def __repr__(self) -> str:
         return self.abi.signature if self.abi else "constructor()"
 
+    def encode_input(self, *args) -> HexBytes:
+        ecosystem = self.provider.network.ecosystem
+        encoded_calldata = ecosystem.encode_calldata(self.abi, *args)
+        return HexBytes(encoded_calldata)
+
+    def decode_input(self, calldata: bytes) -> Tuple[str, Dict[str, Any]]:
+        input_str = ",".join([str(i.internalType or i.type) for i in self.abi.inputs])
+        selector = f"constructor({input_str})"
```

Comment:
Should this be `self.abi.selector`?

Reply:
`selector` is missing from `ConstructorABI` currently

Reply:
https://github.com/ApeWorX/ethpm-types/pull/54

If that releases, I can use it and bump the ethpm-types

Reply:
released https://github.com/ApeWorX/ethpm-types/releases/tag/v0.3.13

Reply:
UGH!!!
https://github.com/ApeWorX/ethpm-types/pull/55